runtimes/runtimes/lsp/router/lspRouter.ts

```diff
@@ -65,4 +80,39 @@ export class LspRouter {
             }
         }
     }
+
+    getUserAgent = (opts: AWSClientInitializationOptions = {}): string => {
```

Comment:
Should this be the responsibility of the Servers? Makes this change a lot simpler and smaller in scope.

Reply:
I considered this at first. Imo the benefit of owning this logic and passing standard computed value is in producing standard value for all Servers. When each server will compute it's own value, there could be discrepancies and it'll be more difficult to manage all different implementations.

Same scalability/management issue as @awschristou mentioned when this logic is owned by Clients, but reversed to the Servers now https://github.com/aws/language-server-runtimes/pull/193#discussion_r1733180679.

As an alternative, we can export this utility function instead of extending `params` with computed `customUserAgent` value for servers to use, then logic for computing stays in Runtimes, but we don't fiddle with params passed down to servers.